### PR TITLE
Update format documentation

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: format.md [hunk 1]
```diff
@@ -34,16 +91,28 @@ Each transition line consists of:
...
-convenient for me, Jon Skeet, as I mostly work on Windows...)
+- The abbreviated name of the zone at this time, e.g. "CST" or "CDT".
+  This is the abbreviation for the period from this transition to
+  the next transition.
+  
+As a complete example, the 2016c data should generate the following
+lines for the `America/La_Paz` zone. 
 
+    America/La_Paz
+    Initially:           -04:32:36 standard LMT
```
Comment (malcolmr) [inline format.md:111]: These spaces after `Initially:` aren't described above.

File: format.md [hunk 2]
```diff
@@ -5,9 +5,63 @@ can be reasonably confident that they have interpreted the source
...
+- `Version`: a version indicator for the data described in the file.
+  Example: `2016d`
+- `Range`: an inclusive range of years covered by transitions in the
+  file. If the file covers all transitions up to a year, the year 1
+  should be used as the start point. Examples: `1-2035`, `1900-2000`.
+- `Generator`: The name of the tool generating the file. Examples:
+  `zdump`, `NodaTime.TzValidate.NodaDump`.
+- `GeneratorUrl`: The URL to visit for more information and/or source
+  code of the tool generating the file.
+- `Body-SHA-256`: A SHA-256 hash of the body of the file.
```
Comment (malcolmr) [inline format.md]: Perhaps "body section".

File: format.md [hunk 3]
```diff
@@ -5,9 +5,63 @@ can be reasonably confident that they have interpreted the source
...
+following header lines are equivalent:
+
+    HeaderKey:HeaderValue
+      HeaderKey  :  HeaderValue
+      
+Common headers
 ----
 
+The intention of the headers section is to provide metadata about
+the body section. A generator may produce any headers it deems
```
Comment (malcolmr) [inline format.md:45]: Are any of these mandatory (or RFC-"SHOULD", perhaps)

File: format.md [hunk 4]
```diff
@@ -22,6 +76,9 @@ The information for each zone consists of:
 - A line per transition strictly before 2035-01-01T00:00:00Z
```
Comment (malcolmr) [inline format.md]: Is this true given Range above?

Comment (jskeet) [inline format.md]: No, I probably need to document the range aspects more carefully.

File: format.md (hunk 3 above)
Comment (jskeet) [inline format.md:45]: Nope. I was tempted to make Format mandatory (and first) to indicate the presence of a header section at all, but as the first line of the body won't include a colon, it's easy to detect the presence of headers anyway.

File: format.md (hunk 2 above)
Comment (jskeet) [inline format.md]: SGTM, will fix.

File: format.md (hunk 1 above)
Comment (jskeet) [inline format.md:111]: Humbug - they're being squashed. They're in the source. Will be more explicit about them anyway, and see whether using `&nbsp;` fixes it.

**Comment by jskeet** [comment]
Right, think I'm now ready to merge - PTAL. I'm going to start implementing anyway :)

**Comment by malcolmr** [comment]
LGTM

**Pull request merged**